controllers/volumereplication_controller.go

```diff
@@ -166,16 +167,22 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		}
 
 	case replicationv1alpha1.Secondary:
-		err := r.markVolumeAsSecondary(volumeHandle, parameters, secret)
+		requeueForResync, err := r.markVolumeAsSecondary(types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, volumeHandle, parameters, secret)
```

Comment:
Can not we use `req.NamespacedName` here instead of types.NamespacedName

Reply:
Moreover, we can pass the instance itself

Reply:
Wouldn't it be better if we fetch the latest instance when we need it?

Reply:
In that case would not it be a problem for example an image state was secondary when the request came while fetching user has changed the image state to the primary. There will be discrepancies as you will have 2 copies of the instance one in your func and the other one in reconcile(). Your func will work according to a new copy of the instance and the reconcile() will work according to the old copy. If we really want to fetch it again we should fetch it into the same pointer IMO.